ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -630,60 +654,62 @@ monitorLedgerStateJudgement PeerSelectionActions{ readLedgerStateJudgement }
-            , hasOnlyBootstrapPeers = False
-            , bootstrapPeersTimeout = Nothing
-            , knownPeers =
-                KnownPeers.delete
-                  nonEstablishedBootstrapPeers
-                  knownPeers
-            , publicRootPeers =
-                PublicRootPeers.difference
-                  publicRootPeers
-                  nonEstablishedBootstrapPeers
-            })
-      return $ \now ->
-        Decision {
-          decisionTrace = [TraceLedgerStateJudgementChanged lsj],
-          decisionJobs  = [],
-          decisionState = st' now
-        }
+      let st' =
+            case lsj of
+              YoungEnough ->
+                let bootstrapPeers = PublicRootPeers.getBootstrapPeers publicRootPeers
+                    nonEstablishedBootstrapPeers =
+                      bootstrapPeers
+                      Set.\\
+                      EstablishedPeers.toSet establishedPeers
+                      Set.\\
+                      (inProgressPromoteCold <> inProgressPromoteWarm)
+                  in \_now -> st { ledgerStateJudgement = lsj,
+                                   knownPeers = KnownPeers.delete
+                                                  nonEstablishedBootstrapPeers
+                                                  knownPeers,
+                                   publicRootPeers = PublicRootPeers.difference
+                                                       publicRootPeers
+                                                       nonEstablishedBootstrapPeers }
+              TooOld ->
+                \now ->
+                  st { ledgerStateJudgement = lsj,
+                       targets = PeerSelectionTargets {
+                         targetNumberOfRootPeers                 = 0,
+                         targetNumberOfKnownPeers                = 0,
+                         targetNumberOfEstablishedPeers          = 0,
+                         targetNumberOfActivePeers               = 0,
+                         targetNumberOfKnownBigLedgerPeers       = 0,
+                         targetNumberOfEstablishedBigLedgerPeers = 0,
+                         targetNumberOfActiveBigLedgerPeers      = 0 },
+                       -- We have to enforce the invariant that the number of root peers is
+                       -- not more than the target number of known peers. It's unlikely in
+                       -- practice so it's ok to resolve it arbitrarily using clampToLimit.
+                       localRootPeers = LocalRootPeers.empty,
+                       bootstrapPeersTimeout = Just (addTime governor_BOOTSTRAP_PEERS_TIMEOUT now),
+                       hasOnlyBootstrapPeers = False }
+
+      return $ decision st' lsj
+
   | otherwise = GuardedSkip Nothing
+  where
+    decision st' lsj now = Decision {
+      decisionTrace = [TraceLedgerStateJudgementChanged lsj],
+        decisionJobs  = [],
+        decisionState = st' now }
```

Comment:
I had some ideas while working on the PR as a whole and was changing this and that in this block, so I did a little refactoring, but then at the end it turned out the ideas I put in here were junk, so I removed them but kept the rest of the refactoring. Turned out I removed too much :( because changing the behavior of bootstrap peers wasn't the goal.